src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -688,6 +688,328 @@ public boolean apply(@Nullable FlowNode input) {
         });
     }
 
+    @Issue("JENKINS-36547")
+    @Test public void reuseNodeFromPreviousRun() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                for (int i = 0; i < 5; ++i) {
+                    story.j.createOnlineSlave();
+                }
+
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "demo");
+                p.setDefinition(new CpsFlowDefinition("node {echo \"ran node block\"}", true));
+                story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+                List<String> log = p.getLastBuild().getLog(10);
+                String firstNode = "";
+                for (String line: log) {
+                    if (line.contains("Running on")) {
```

Comment:
More robust to look for `WorkspaceAction` here.

Reply:
@jglick: I got two questions about this proposal:

1. How to find the node name or something similar reliable to check for?
Maybe I'm doing something wrong. However when looking at the `WorkspaceAction` here the `getNode()` method returns an empty string. I'd have assumed that that one would hold the name of the node being used.

```java
    private List<WorkspaceAction> getWorkspaceActions(WorkflowRun workflowRun) {
        FlowGraphWalker walker = new FlowGraphWalker(workflowRun.getExecution());
        List<WorkspaceAction> actions = new ArrayList<WorkspaceAction>();
        for (FlowNode n : walker) {
            WorkspaceAction a = n.getAction(WorkspaceAction.class);
            if (a != null) {
                actions.add(a);
            }
        }
        return actions;
    }
```

in the test I have:
```java
                WorkflowRun run = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
                List<WorkspaceAction> workspaceActions = getWorkspaceActions(run);

                String firstNode = null;
                for (WorkspaceAction action: workspaceActions) {
                    if (firstNode == null) {
                        firstNode = action.getNode();
                        firstNode = action.getWorkspace().toComputer().getName();
                    } else {
                        assertEquals(action.getNode(), firstNode);
                    }
                }
```

Both `action.getNode()` as well as `action.getWorkspace().toComputer().getName()` return an empty string here (there's only one action retrieved). Maybe the workspace action retrieved is running on the master?

Not sure whether I am doing something wrong. I'd appreciate any hints that point me to the right direction.

2. What to do with the remaining tests
The other tests currently are searching for the node block where a particular echo step has been executed in. I was thinking about writing some file into the workspace in order to check whether the file is still there after restarting the job. But somehow this sounds little bit too complicated. Would you go for parsing the flow nodes here as well?